### ISSUE TITLE
detect m3u8 quality in exoplayer

### ISSUE THREAD

**Issue by supriyatambe131995** [google/ExoPlayer#6375]
Before filing a feature request:
-----------------------
- Search existing open issues, specifically with the label ‘enhancement’:
  https://github.com/google/ExoPlayer/labels/enhancement
- Search existing pull requests: https://github.com/google/ExoPlayer/pulls

When filing a feature request:
-----------------------
Fill out the sections below, leaving the headers but replacing the content. If
you're unable to provide certain information, please explain why in the relevant
section. We may close issues if they do not include sufficient information.

### [REQUIRED] Use case description
Describe the use case or problem you are trying to solve in detail. If there are
any standards or specifications involved, please provide the relevant details.

### Proposed solution
A clear and concise description of your proposed solution, if you have one.

### Alternatives considered
A clear and concise description of any alternative solutions you considered,
if applicable.

**Comment by andrewlewis** [comment]
There isn't enough information here for us to help. Please give a through description of the feature you're looking for.

**Comment by saurabhdtu** [comment]
using exoplayer 2.10.2
@andrewlewis  I am also facing problem regarding bandwidth and track adaptability. Here's the issue
I am  playing a m3u8 file using the following code
` 
TrackSelector trackSelector = new DefaultTrackSelector(videoTrackSelectionFactory);
simpleExoPlayer = ExoPlayerFactory.newSimpleInstance(this);
hlsDataSource = new DefaultDataSourceFactory(this,
Util.getUserAgent(this, "Rooter"));
simpleExoPlayer.setRepeatMode(Player.REPEAT_MODE_ONE);
HlsMediaSource hlsMediaSource = new HlsMediaSource.Factory(hlsDataSource).createMediaSource(uri);
concatenatingMediaSource.addMediaSource(hlsMediaSource);
simpleExoPlayer.prepare(concatenatingMediaSource);
`
The track selection based on network speed is only working when the video from the playlist changes. When the first track loads the bandwidth of the previous m3u8 file is used and doesn't change till the file ends or I am seeking the file to a different unbuffered timestamp.
Basically the exoplayer keeps on buffering with the high bitrate unless untill the exoplayer is seeked to m3u8 file.

**Comment by saurabhdtu** [comment]
Any update on this?

**Comment by stevemayhew** [comment]
@saurabhdtu The default bandwidth algorithm in ExoPlayer (https://github.com/google/ExoPlayer/blob/85c10b02567b66ecfaaaacfd0ba9389b41bf919e/library/core/src/main/java/com/google/android/exoplayer2/upstream/DefaultBandwidthMeter.java#L48) initially considers the interface bandwidth (on a mobile or wireless device this makes a lot of sense and leads to faster adaption on source switches).

If this is not the desired behavior for your use-case you could construct the player passing your own bandwidth metering using this method: https://github.com/google/ExoPlayer/blob/44293e8f45fc62ff918001b4425c44d0ded5ffc9/library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerFactory.java#L433

Changing the sliding window weight affects how quickly old bandwidth samples are aged out.

as @andrewlewis said, give more details on your use case and I'm confident the existing ExoPlayer can work.

**Comment by ojw28** [comment]
Closing due to lack of a clear description of the issue.
